packages/subgraph/schema.graphql

```diff
@@ -107,70 +107,102 @@ type Dao @entity {
+  installations: [PluginInstallation!] @derivedFrom(field: "appliedPluginRepo")
+}
+
 type PluginSetup @entity(immutable: true) {
   id: ID! # contract address
   versions: [PluginVersion!]! @derivedFrom(field: "pluginSetup")
-  installations: [Plugin!]! @derivedFrom(field: "pluginSetup")
 }
 
 type PluginVersion @entity(immutable: true) {
-  id: ID! # pluginRepo + version index
+  id: ID! # pluginRepo + release + build
   pluginRepo: PluginRepo!
-  semanticVersion: [BigInt!]!
-  pluginSetup: PluginSetup!
-  contentURI: Bytes!
+  pluginSetup: PluginSetup
+  release: Int!
+  build: Int!
+  metadata: Bytes!
+  preparations: [PluginPreparation!]! @derivedFrom(field: "pluginVersion")
+  installations: [PluginInstallation!] @derivedFrom(field: "appliedVersion")
+}
+
+type PluginPermission @entity(immutable: true) {
+  id: ID! # PluginPreparation + operation + where + who + permissionId
+  pluginPreparation: PluginPreparation!
+  operation: PermissionOperation!
+  where: Bytes!
+  who: Bytes!
+  condition: Bytes
+  permissionId: Bytes!
 }
 
-type PluginRepo @entity(immutable: true) {
-  id: ID! # address
-  subdomain: String!
-  versions: [PluginVersion!]! @derivedFrom(field: "pluginRepo")
+enum PermissionOperation {
+  Grant
+  Revoke
+  GrantWithCondition
 }
 
-type PluginHelper @entity(immutable: true) {
-  id: ID! # address
-  plugin: Plugin!
+type PluginPreparation @entity(immutable: true) {
+  id: ID! # psp setupId + installationId
+  creator: Bytes!
+  dao: Dao!
+  setupId: Bytes! # the psp setupId
+  pluginRepo: PluginRepo!
+  pluginVersion: PluginVersion!
+  data: Bytes
+  plugin: PluginInstallation!
+  helpers: [Bytes!]!
+  permissions: [PluginPermission!]! @derivedFrom(field: "pluginPreparation")
+  type: PluginPreparationType!
 }
 
-type Plugin @entity {
-  "TODO: should do `implements IPlugin` once plugin is seperated from core"
-  id: ID! # address
+type PluginInstallation @entity {
+  id: ID! # psp installationId
   dao: Dao!
-  pluginSetup: PluginSetup!
-  data: Bytes!
-  helpers: [PluginHelper!]! @derivedFrom(field: "plugin")
-  state: PluginState!
-  sender: String!
+  appliedPreparation: PluginPreparation
+  appliedSetupId: Bytes # The setupId of the application see PSP documentation for more info
+  appliedVersion: PluginVersion # Stored to track installations in the different plugin versions
+  appliedPluginRepo: PluginRepo # Stored to track installation in the different plugin repos
+  preparations: [PluginPreparation!]! @derivedFrom(field: "plugin")
+  state: PluginPreparationState!
 }
 
-enum PluginState {
+enum PluginPreparationType {
+  Installation
+  Update
+}
+
+enum PluginPreparationState {
   InstallationPrepared
   Installed
   UpdatePrepared
   UninstallPrepared
   Uninstalled
 }
 
-type DaoPlugin @entity {
-  id: ID! # Dao + Plugin
-  plugin: IPlugin!
-  dao: Dao!
-}
-
-interface IPlugin {
-  id: ID! # address
-  daos: [DaoPlugin!] @derivedFrom(field: "plugin")
+interface IPluginInstallation {
```

Comment:
why it is called `IPluginInstallation` and not implemented by `PluginInstallation`?
may be somthing is off. 
perhaps it would be fine to keep it as IPlugin, and add pluginAddress to it, check my comment below.

Reply:
added it. I didn't see the point of having it implemented because  `IPluginInstallation` isn't complete

Reply:
yes, 

is this interface meants to be removed once plugins go out of the core?
if yes, i would add TODO explaining ...etc